firebase/functions/index.js

```diff
@@ -27,118 +45,140 @@ exports.resultCounter = functions.database.ref('/v2/results/{projectId}/{groupId
-    const userRef = admin.database().ref('/v2/users/' + context.params.userId)
-    const userContributionRef = userRef.child('contributions/' + context.params.projectId)
-
-    // Counter for User contributions across Projects
-    const totalGroupContributionCount = userRef.child('groupContributionCount').transaction((currentCount) => {
-        return currentCount + 1
-    });
-    promises.push(totalGroupContributionCount)
-
-    const totalTaskContributionCount = groupRef.child('numberOfTasks').once('value')
-        .then((dataSnapshot) => {
-            const numberOfTasks = dataSnapshot.val()
-            return numberOfTasks
-        })
-        .then((numberOfTasks) => {
-            return userRef.child('taskContributionCount').transaction((currentCount) => {
-                return currentCount + numberOfTasks
-            })
-        })
-    promises.push(totalTaskContributionCount)
-
-    // Counter for User contributions on a distict Project
-    const groupContributionCount = userContributionRef.child('groupContributionCount').transaction((currentCount) => {
-        return currentCount + 1
-    });
-    promises.push(groupContributionCount)
-
-    const taskContributionCount = groupRef.child('numberOfTasks').once('value')
-        .then((dataSnapshot) => {
-            const numberOfTasks = dataSnapshot.val()
-            return numberOfTasks
-        })
-        .then((numberOfTasks) => {
-            return userContributionRef.child('taskContributionCount').transaction((currentCount) => {
-                return currentCount + numberOfTasks
-            })
-        })
-    promises.push(taskContributionCount)
-
-    const contributionTime = userContributionRef.child(context.params.groupId).once('value')
+    /*
+        Check if this user has submitted a results for this group already.
+        If no result has been submitted yet, set userId in v2/groupsUsers.
+        Then set this group contribution in the user profile.
+        Update overall taskContributionCount and project taskContributionCount in the user profile
+        based on the number of results submitted and the existing count values.
+    */
+    updateValues = groupUsersRef.child(context.params.userId).once('value')
         .then((dataSnapshot) => {
             if (dataSnapshot.exists()) {
                 console.log('group contribution exists already. user: '+context.params.userId+' project: '+context.params.projectId+' group: '+context.params.groupId)
                 return null
             }
             else {
-                const data = {
-                    'startTime': result['startTime'],
-                    'endTime': result['endTime']
+                const numberOfTasks = Object.keys( result['results'] ).length
+                return {
+                    userContribution: userContributionRef.child(context.params.groupId).set(true),
```

Comment:
@laurentS do you see any problem here?

We are not using the data that is set in the user contributions in Firebase as far as I know. The app just checks for the groupId in `v2/users/{userId}/contributions/{projectId}/{groupId}`. Is this correct?

Reply:
yes, none of that is used in the app